tests/test_interpolation.py

```diff
@@ -250,6 +250,30 @@ def test_resolver_cache_2():
         OmegaConf.clear_resolvers()
 
 
+def test_resolver_that_allows_any_char():
+    def env_or_default(x):
+        split = x.split(",")
+        if len(split) > 1:
+            return os.getenv(split[0], split[1])
+        return os.getenv(split[0])
```

Comment:
```python
    def env_or_default(name, def = None):
        if def is not None:
            return os.getenv(name, def)
        return os.getenv(name)

    c = OmegaConf.create(
      dict(
        default_env_set="${env_or_default:arg1,env_name,env_default}",
      )
    )
```

Reply:
something like this.
It's more work but more useful.

Reply:
Looks good, I'll have a crack later this week

Reply:
Should we then make `,` a special token that denotes an argument list?

Reply:
I think this is reasonable, but it would be good to also support \, to escape it and pass a literal ,.
another reasonable thing is to strip whitespace around arguments (and escape whitespace with '\ '.

Reply:
This can also naturally support 0 argument calls:
`${foo:}`